### PR TITLE
Parsing image version.

### LINKED ISSUE #465: fix image version discrepancy
should be v3.10.34, not v3.10.14

### PR DESCRIPTION
**Pull request by kwoodson**
Parsing image version to determine correct version string.

fixes #465

### CODE REVIEW

**Comment by kwoodson** [comment]
@jim-minter, would you prefer a hard coded version string or the regexp?

File: pkg/config/images.go [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
+	"regexp"
 	"strings"
 
 	acsapi "github.com/openshift/openshift-azure/pkg/api"
 )
 
+func imageVersion(incVersion string) string {
+	// Version should be in xxx.xx.xxxxxxxx format
+	// e.g: 310.14.20180101
+	rxVersion := regexp.MustCompile(`^([0-9]){1}([0-9]{1,3})` +
```
Comment (jim-minter) [inline pkg/config/images.go]: use strings.Split

File: pkg/config/images.go [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
 import (
+	"fmt"
 	"os"
+	"regexp"
 	"strings"
 
 	acsapi "github.com/openshift/openshift-azure/pkg/api"
 )
 
+func imageVersion(incVersion string) string {
```
Comment (jim-minter) [inline pkg/config/images.go]: unit test

Comment (jim-minter) [inline pkg/config/images.go]: `func openShiftVersion(imageVersion string) string {`

File: pkg/config/images.go (hunk 1 above)
Comment (kwoodson) [reply pkg/config/images.go]: I looked into string.split but wasn't aware of a clean way to divide the numbers exactly how we wanted them.
`310.14.20180101` to `3.10.14`

If you have an example that'd be great.

File: pkg/config/images_test.go [hunk 3]
```diff
@@ -6,6 +6,40 @@ import (
...
+		testVersion string
+		testResult  string
+		testError   string
+	}{
+		"valid version string": {
+			testVersion: "310.14.20180101",
+			testResult:  "v3.10.14",
+		},
+		"invalid version string length": {
+			testVersion: "310.14.20180101",
```
Comment (kwoodson) [inline pkg/config/images_test.go]: Need to add more length here to cause the version string to be invalid.

File: pkg/config/images.go [hunk 4]
```diff
@@ -1,12 +1,24 @@
...
 	"os"
 	"strings"
 
 	acsapi "github.com/openshift/openshift-azure/pkg/api"
 )
 
+// openshiftVersion converts a VM image version (e.g. 310.14.20180101) to an
+// openshift container image version (e.g. v3.10.14)
+func openShiftVersion(imageVersion string) (string, error) {
+	parts := strings.Split(imageVersion, ".")
```
Comment (kwoodson) [inline pkg/config/images.go]: I understand how you wanted to use split and use the returned string for `part[0]` and use array index for the version.

**Pull request merged**